docs/Testing.md

```diff
@@ -48,52 +45,20 @@ Before committing a large change to a compiler (especially a language change),
 or API changes to the standard library, it is recommended to run validation
 test suite, via ``utils/build-script --validation-test``.
 
-Although it is not recommended for day-to-day contributions, it is also
-technically possible to execute the tests directly via CMake. For example, if you have
-built Swift products at the directory ``build/Ninja-ReleaseAssert/swift-macosx-x86_64``,
-you may run the entire test suite directly using the following command:
-
-```
-  cmake --build build/Ninja-ReleaseAssert/swift-macosx-x86_64 -- check-swift-macosx-x86_64
-```
-
-Note that ``check-swift`` is suffixed with a target operating system and architecture.
-Besides ``check-swift``, other targets are also available. Here's the full list:
+Using ``utils/build-script`` will rebuild all targets which can add substantial
+time to a debug cycle.
 
-* ``check-swift``: Runs tests from the ``${SWIFT_SOURCE_ROOT}/test`` directory.
-* ``check-swift-only_validation``: Runs tests from the ``${SWIFT_SOURCE_ROOT}/validation-test`` directory.
-* ``check-swift-validation``: Runs the primary and validation tests, without the long tests.
-* ``check-swift-only_long``: Runs long tests only.
-* ``check-swift-all``: Runs all tests (primary, validation, and long).
-* ``SwiftUnitTests``: Builds all unit tests.  Executables are located under
-  ``${SWIFT_BUILD_ROOT}/unittests`` and must be run individually.
+#### Using lit.py
 
-For every target above, there are variants for different optimizations:
-
-* the target itself (e.g., ``check-swift``) -- runs all tests from the primary
-  testsuite.  The execution tests are run in ``-Onone`` mode.
-* the target with ``-optimize`` suffix (e.g., ``check-swift-optimize``) -- runs
-  execution tests in ``-O`` mode.  This target will only run tests marked as
-  ``executable_test``.
-* the target with ``-optimize-unchecked`` suffix (e.g.,
-  ``check-swift-optimize-unchecked``) -- runs execution tests in
-  ``-Ounchecked`` mode. This target will only run tests marked as
-  ``executable_test``.
-* the target with ``-executable`` suffix (e.g.,
-  ``check-swift-executable-iphoneos-arm64``) -- runs tests marked with
-  ``executable_test`` in ``-Onone`` mode.
-* the target with ``-non-executable`` suffix (e.g.,
-  ``check-swift-non-executable-iphoneos-arm64``) -- runs tests not marked with
-  ``executable_test`` in ``-Onone`` mode.
-
-If more control is required (e.g. to manually run certain tests), you can invoke
-LLVM's lit.py script directly. For example:
+Using `lit.py` directly can provide more control and faster feedback to your
+development cycle. To invoke LLVM's `lit.py` script directly, it must be
+configured to use your local build directory. For example:
 
 ```
     % ${LLVM_SOURCE_ROOT}/utils/lit/lit.py -sv ${SWIFT_BUILD_ROOT}/test-iphonesimulator-i386/Parse/
```

Comment:
This is worth hyperlinking to from the GettingStarted guide, as it's one of the first things I tell new contributors once they're building and running. Otherwise, it's rather buried.